src/stock.py

```diff
@@ -0,0 +1,48 @@
+class Stock:
+    name: str
+    shares: int
+    price: float
+    change: float
+
+    def __init__(self, name: str, shares: int, price: float) -> None:
+        self.name = name
+        self.shares = shares
+        self.price = price
+        self.change: float
+
+    def cost(self) -> float:
+        """
+        Calculates the cost of the stock.
+
+        :return: Returns the cost of the stock
+        """
+        return self.shares * self.price
+
+    def sell(self, nshares: int) -> None:
+        """
+        Subtracts shares.
+
+        :param nshares: Number of shares to sell
+        """
+        self.shares -= nshares
+
+    def __eq__(self, other: object) -> bool:
+        """
+        Tests equivalence of two Stock objects.
+
+        :param other: Stock object to compare to
+        :return: Returns True if self and other have the same attributes otherwise False
+        """
+        if not isinstance(other, Stock):
+            return NotImplemented
+
+        return all(
```

Comment:
You can simply this by using `__dict__`:
```suggestion
        return self.__dict__ == other.__dict__
```

The advantage of `__dict__` is that you don't have to update the `__eq__` if you are adding a new attribute (assuming this new attribute is mandatory to define equivalence between 2 stocks)

Reply:
Changed.